### PR TITLE
Issues/#107 SHA-3 conversion in signature 

### PR DESCRIPTION
**Pull request by macterra**
Functional tests:
```
Total Passed: 68
Total Failed: 6
Number of tests: 75
```

### CODE REVIEW

File: src/test/bip32_tests.cpp [hunk 1]
```diff
@@ -87,6 +89,61 @@ TestVector test3 =
...
+  TestVector("4b381541583be4423346c643850da4b320e46a87ae3d2a4e6da11eba819cd4acba45d239319ac14f863b8d5ab5a0d0c64d2e8a1e7d1457df2e5a3c51c73235be")
+    ("xpub661MyMwAqRbcH67oyNT8SMJBZp8sovtbjs8k555MCy9r5ihLQSCJBJstFiqTzN5Ek6EAgJNDLWaMXbwiVU7aGfyykF6qYqnxJpvP5EsJuQJ",
+     "xprv9s21ZrQH143K4c3LsLv85DMT1nJPQUAkNeD9GgfjedcsCvNBrtt3dWZQQTokDHLQv1NKeid3yoJCdk2hCATJjvhrM5DjyK332pU133fENvT",
+    0x80000000)
+    ("xpub69aqk6TAsLPaR9QG9k31njketbWD6RZVHQ7FRNJwFLAqyUN6EUZvByGnnx6CKUq5P7oaUPmCETDkh7zheWgm6wWjxmtNtdeN7czfLww8nPk",
+     "xprv9vbVLavH2xqHCfKo3iW1RbovLZfigxqdvBBecyuKgzds6g2wgwFfeAxJwfGJoA41HW2A14JZfv7vboP74HjrBMDFwqLJaJ2wz5ADLzeNUrr",
+     0);
+
+#endif // END_BUILD
+
```
Comment (r-soltani) [inline src/test/bip32_tests.cpp]: What library was used to generate these keys?

File: src/pubkey.h [hunk 2]
```diff
@@ -148,13 +148,21 @@ class CPubKey
     //! Get the KeyID of this public key (hash of its serialization)
```
Comment (r-soltani) [inline src/pubkey.h]: Please add the copyright

File: src/test/bip32_tests.cpp [hunk 3]
```diff
@@ -36,6 +36,8 @@ struct TestVector {
     }
```
Comment (r-soltani) [inline src/test/bip32_tests.cpp]: please add the copyright

**Comment by macterra** [comment]
Just fyi, I reverted most of the wallet conversion, will be deferred to a separate PR for #118

**Review by ealefad — APPROVED** [review]

File: src/wallet/wallet.cpp [hunk 4]
```diff
@@ -4248,7 +4253,13 @@ CTxDestination CWallet::AddAndGetDestinationForScript(const CScript& script, Out
     case OUTPUT_TYPE_P2SH_SEGWIT:
     case OUTPUT_TYPE_BECH32: {
         WitnessV0ScriptHash hash;
+#ifdef BUILD_BTC
         CSHA256().Write(script.data(), script.size()).Finalize(hash.begin());
+#else  // BUILD_EQB
+        CSHA256().Write(script.data(), script.size()).Finalize(hash.begin());
+        // EQB_TODO revisit as time permits
+        //SHA3().Write(script.data(), script.size()).Finalize(hash.begin());
```
Comment (r-soltani) [inline src/wallet/wallet.cpp:4261]: We are not able to use SHA3 here?

Comment (macterra) [inline src/wallet/wallet.cpp:4261]: This was causing a functional test failure that will be fixed with the PR for #118

**Review by r-soltani — APPROVED** [review]

File: src/wallet/wallet.cpp (hunk 4 above)
Comment (ealefad) [reply src/wallet/wallet.cpp:4261]: @r-soltani here's some history of this https://github.com/Equibit/equibit-core/issues/141

**Pull request merged**